nwbwidgets/controllers/time_window_controllers.py

```diff
@@ -1,101 +1,102 @@
-from abc import abstractmethod
+from typing import Tuple, Optional
 
-from ipywidgets import HBox, Layout, ValueWidget, link, widgets
+from ipywidgets import Layout, ValueWidget, link, widgets
 from ipywidgets.widgets.widget_description import DescriptionWidget
 
+from .basecontroller import BaseController
 
-class WindowController(HBox, ValueWidget, DescriptionWidget):
+
+class WindowController(BaseController, ValueWidget, DescriptionWidget):
     def __init__(
-        self, vmin, vmax, start_value=(None, None), description="window (s)", orientation="horizontal", **kwargs
+        self,
+        vmin: int,
+        vmax: int,
+        start_value: Optional[Tuple[Optional[int], Optional[int]]] = (None, None),
+        description: str = "window (s)",
+        orientation: str = "horizontal",  # TODO: should be Literal["horizontal", "vertical"]
+        **kwargs,
     ):
-
-        if orientation not in ("horizontal", "vertical"):
-            ValueError("Unrecognized orientation: {}".format(orientation))
-
-        super().__init__()
         self.vmin = vmin
         self.vmax = vmax
         self.start_value = start_value
         self.description = description
         self.orientation = orientation
 
+        super().__init__()
+
+    def setup_attributes(self):
         if self.orientation == "horizontal":
             self.to_start_button = widgets.Button(description="◀◀", layout=Layout(width="65px"))
             self.backwards_button = widgets.Button(description="◀", layout=Layout(width="40px"))
             self.forward_button = widgets.Button(description="▶", layout=Layout(width="40px"))
             self.to_end_button = widgets.Button(description="▶▶", layout=Layout(width="65px"))
-        else:  # vertical
+        elif self.orientation == "vertical":
             self.to_end_button = widgets.Button(description="▲▲", layout=Layout(width="50px"))
             self.forward_button = widgets.Button(description="▲", layout=Layout(width="50px"))
             self.backwards_button = widgets.Button(description="▼", layout=Layout(width="50px"))
             self.to_start_button = widgets.Button(description="▼▼", layout=Layout(width="50px"))
+        else:
+            raise ValueError(
+                f"Unrecognized orientation '{self.orientation}', should be either 'horizontal' or 'vertical'"
+            )
 
-        self.to_start_button.on_click(self.move_start)
-        self.backwards_button.on_click(self.move_down)
-        self.forward_button.on_click(self.move_up)
-        self.to_end_button.on_click(self.move_end)
-
-    # @abstractmethod
-    # @property
-    # def value(self):
-    #    """Must be window in seconds"""
-    #    pass
-
-    @abstractmethod
-    def get_children(self):
-        pass
-
-    @abstractmethod
```

Comment:
Note that the usage of `abstractmethod` here does nothing since the class does not inherit from `ABC`. Normally this would prevent instantiation of the class entirely with an informative error telling the user to define those methods.

Swapped to raising `NotImplementedError`s on function call instead.